lifecycle-operator/webhooks/pod_mutator/handlers/objectmeta.go

```diff
@@ -0,0 +1,121 @@
+package handlers
+
+import (
+	"fmt"
+	"hash/fnv"
+	"strings"
+
+	apicommon "github.com/keptn/lifecycle-toolkit/lifecycle-operator/apis/lifecycle/v1alpha3/common"
+	operatorcommon "github.com/keptn/lifecycle-toolkit/lifecycle-operator/common"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func GetLabelOrAnnotation(resource *metav1.ObjectMeta, primaryAnnotation string, secondaryAnnotation string) (string, bool) {
+
+	if resource.Annotations[primaryAnnotation] != "" {
+		return resource.Annotations[primaryAnnotation], true
+	}
+
+	if resource.Labels[primaryAnnotation] != "" {
+		return resource.Labels[primaryAnnotation], true
+	}
+
+	if secondaryAnnotation == "" {
+		return "", false
+	}
+
+	if resource.Annotations[secondaryAnnotation] != "" {
+		return resource.Annotations[secondaryAnnotation], true
+	}
+
+	if resource.Labels[secondaryAnnotation] != "" {
+		return resource.Labels[secondaryAnnotation], true
+	}
+	return "", false
+}
+
+func getWorkloadName(meta *metav1.ObjectMeta, applicationName string) string {
+	workloadName, _ := GetLabelOrAnnotation(meta, apicommon.WorkloadAnnotation, apicommon.K8sRecommendedWorkloadAnnotations)
+	return operatorcommon.CreateResourceName(apicommon.MaxK8sObjectLength, apicommon.MinKeptnNameLen, applicationName, workloadName)
+}
+
+func getAppName(meta *metav1.ObjectMeta) string {
+	var applicationName string
+	if !isAppAnnotationPresent(meta) {
+		applicationName, _ = GetLabelOrAnnotation(meta, apicommon.WorkloadAnnotation, apicommon.K8sRecommendedWorkloadAnnotations)
+	} else {
+		applicationName, _ = GetLabelOrAnnotation(meta, apicommon.AppAnnotation, apicommon.K8sRecommendedAppAnnotations)
+	}
+	return strings.ToLower(applicationName)
+}
+
+func getAnnotations(objMeta *metav1.ObjectMeta, annotationKey string) []string {
```

Comment:
the name of the function might suggest that this just returns the annotations from the objMeta - maybe something like `getValuesFromAnnotation` might make this a bit clearer, together with a short comment